src/account/UpgradeableModularAccount.sol

```diff
@@ -545,9 +522,8 @@ contract UpgradeableModularAccount is
     function _doCachedPostExecHooks(PostExecToRun[] memory postHooksToRun) internal {
         uint256 postHooksToRunLength = postHooksToRun.length;
         for (uint256 i = postHooksToRunLength; i > 0;) {
-            unchecked {
-                --i;
-            }
+            // Decrement here, instead of in the loop body, to handle the case where length is 0.
```

Comment:
To be a bit more clear, this is to be able to index into `postHooksToRun[i]` below instead of `postHooksToRun[i-1]`. Very minor optimization

Reply:
Ah I see. I tried setting the starting value of i to `postHooksToRunLength` - 1 at first, but this causes an underflow at length = 0, so I added this comment to indicate usage.